compiler/rustc_builtin_macros/src/test_harness.rs

```diff
@@ -290,7 +290,7 @@ fn mk_main(cx: &mut TestCtxt<'_>) -> P<ast::Item> {
     let mut test_runner = cx
         .test_runner
         .clone()
-        .unwrap_or(ecx.path(sp, vec![test_id, Ident::from_str_and_span(runner_name, sp)]));
+        .unwrap_or_else(|| ecx.path(sp, vec![test_id, Ident::from_str_and_span(runner_name, sp)]));
```

Comment:
Is the `path` function complex or is the readability more important?

Reply:
that's one seems fine, esp because we are allocating a vec here

Reply:
Ok, so `unwrap_or_else` it is.